### PR TITLE
Switch to “intents” API

### PR DESCRIPTION
**Pull request by katzmo**

### CODE REVIEW

File: src/CreditCardForm.php [hunk 1]
```diff
@@ -35,21 +35,21 @@ public function form(array $form, array &$form_state, \Payment $payment) {
     );
 
     drupal_add_js($settings, 'setting');
+    drupal_add_js('https://js.stripe.com/v3/', 'external');
```
Comment (torotil) [inline src/CreditCardForm.php]: In general I’d prefer using #attached over direct use of `drupal_add_js()`.

**Review by torotil — COMMENTED** [review]
… and many `phpcs` issues.

File: src/CreditCardController.php [hunk 2]
```diff
@@ -45,141 +68,182 @@ function validate(\Payment $payment, \PaymentMethod $method, $strict) {
...
-      ));
+        'currency' => $payment->currency_code,
+      ]);
+    }
+    // SetupIntent: Save card details for later use without initial payment.
+    if ($this->filterRecurringLineItems($payment, TRUE)) {
+      return SetupIntent::create();
+    }
+    // TODO: What now? Exception for "nothing to pay for"?
+  }
```
Comment (torotil) [inline src/CreditCardController.php]: This means the payment method can’t deal with mixed line items, right? It always uses a single intent for the whole execution.

File: src/CreditCardController.php [hunk 3]
```diff
@@ -45,141 +68,182 @@ function validate(\Payment $payment, \PaymentMethod $method, $strict) {
...
-    return \Stripe\Charge::create(array(
-        'customer' => $customer->id,
+  /**
+   * Create a new payment intent using the API.
+   */
+  public function createIntent($payment) {
+    // PaymentIntent: Make a payment immediately.
+    if ($this->filterRecurringLineItems($payment, FALSE)) {
+      return PaymentIntent::create([
         'amount'   => $this->getTotalAmount($payment),
```
Comment (torotil) [inline src/CreditCardController.php]: This gets the total of all line items. Even if some of them are recurring.

**Review by torotil — CHANGES_REQUESTED** [review]

File: src/CreditCardController.php (hunk 2 above)
Comment (katzmo) [reply src/CreditCardController.php]: No, it’s fine to use a PaymentIntent for subscriptions too.

File: src/CreditCardController.php (hunk 3 above)
Comment (katzmo) [reply src/CreditCardController.php]: That’s right, `getTotalAmout()` should only sum up one-off line-items!

File: src/CreditCardController.php (hunk 2 above)
Comment (torotil) [reply src/CreditCardController.php]: Ok, after discussion: PaymentIntents can (and should) be used whenever there is at least one non-recurring line-item.

**Review by torotil — APPROVED** [review]

**Pull request merged**